File: Native_Kotlin/app/src/main/java/com/example/activityautostart/selector/SelectorHolder.kt
package com.example.activityautostart.selector

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView
import com.example.activityautostart.data.DataModel
import com.example.movch.activityautostart.R
import kotlinx.android.synthetic.main.holder_selector_card.view.*

class SelectorHolder(itemView: View, private val controller: SelectorController) : RecyclerView.ViewHolder(itemView) {
    fun init(dataModel: DataModel) {
        if (dataModel.isChecked) itemView.selectorMark.visibility = View.VISIBLE
        else itemView.selectorMark.visibility = View.INVISIBLE

        val selectorTextView: TextView = itemView.findViewById(R.id.selectorCheckText)
        selectorTextView.text = dataModel.id


        if (!itemView.selectorCheckBox.hasOnClickListeners()) {
            itemView.selectorCheckBox.setOnClickListener { controller.changeSelection(selectorTextView.text.toString()) }
        }
    }
}

File: Xamarin_CSharp/ActivityAutostart/Recycler/SelectorHolder.cs
using System;
using Android.Support.V7.Widget;
using Android.Views;
using Android.Widget;
using Portable.Data;

namespace ActivityAutostart.Recycler
{
    public class SelectorHolder : RecyclerView.ViewHolder
    {
        public event Action<string> OnSelectorClick;
        private string _id;

        public SelectorHolder(View itemView) : base(itemView)
        {
        }

        public void Init(DataModel dataModel)
        {
            var selectorMark = ItemView.FindViewById<ImageView>(Resource.Id.selectorMark);
            selectorMark.Visibility = dataModel.IsChecked ? ViewStates.Visible : ViewStates.Invisible;

            var selectorTextView = ItemView.FindViewById<TextView>(Resource.Id.selectorCheckText);
            selectorTextView.Text = dataModel.Id;

            _id = dataModel.Id;

            var holderLl = ItemView.FindViewById<LinearLayout>(Resource.Id.selectorCheckBox);
            if (!holderLl.HasOnClickListeners)
            {
                holderLl.Click += SelectorClicked;
            }
        }

        private void SelectorClicked(object sender, EventArgs e)
        {
            OnSelectorClick?.Invoke(_id);
        }
    }
}